### PR TITLE
[ti_crowdstrike] Add the ECS mappings to be useful for threat Intel rules

### PR DESCRIPTION
**Pull request by mohitjha-elastic**
Type of change
- Enhancement

## What does this PR do?
**Add the ECS mappings to be useful for threat Intel rules**
Customer provided the below lists of ECS mappings that are required to be added to be useful for threat Intel rules, hence we supported the below list of ECS mappings in this PR:

**URL:**
threat.indicator.url.full 

**IP:**
threat.indicator.ip

**Hash:**
threat.indicator.file.hash.md5
threat.indicator.file.hash.sha1
threat.indicator.file.hash.sha256

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.12.0

## How to test this PR locally
Clone integrations repo.
Install the elastic package locally.
Start the elastic stack using the elastic package.
Move to integrations/packages/ti_crowdstrike directory.
Run the following command to run tests.
`elastic-package test -v`

## Related issues

- Closes https://github.com/elastic/integrations/issues/9455

## Automated Test
[test-ti_crowdstrike-0.5.4.log](https://github.com/elastic/integrations/files/14768647/test-ti_crowdstrike-0.5.4.log)

### CODE REVIEW

File: packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -347,20 +336,41 @@ processors:
...
+      tag: set_threat_indicator_file_hash_sha256
       value: '{{{ti_crowdstrike.intel.value}}}'
       if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha256') && ctx.ti_crowdstrike?.intel?.value != null
   - set:
-      field: file.hash.md5
-      tag: set_file_hash_md5
+      field: threat.indicator.file.hash.sha1
+      tag: set_threat_indicator_file_hash_sha1
+      value: '{{{ti_crowdstrike.intel.value}}}'
+      if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha1') && ctx.ti_crowdstrike?.intel?.value != null
```
Comment (efd6) [inline packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml:347]:
```suggestion
      if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.value != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha1')
```
Removes unnecessary null-safe operators and puts cheaper checks first; similar changes below.

File: packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -347,20 +336,41 @@ processors:
...
+  - set:
+      field: threat.indicator.file.hash.md5
+      tag: set_threat_indicator_file_hash_md5
       value: '{{{ti_crowdstrike.intel.value}}}'
       if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_md5') && ctx.ti_crowdstrike?.intel?.value != null
+  - set:
+      field: threat.indicator.url.full
```
Comment (kcreddy) [inline packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml:354]: Can you apply `uri_parts` processor on this URL?

File: packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -347,20 +336,41 @@ processors:
...
       if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha256') && ctx.ti_crowdstrike?.intel?.value != null
   - set:
-      field: file.hash.md5
-      tag: set_file_hash_md5
+      field: threat.indicator.file.hash.sha1
+      tag: set_threat_indicator_file_hash_sha1
+      value: '{{{ti_crowdstrike.intel.value}}}'
+      if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha1') && ctx.ti_crowdstrike?.intel?.value != null
+  - set:
+      field: threat.indicator.file.hash.md5
```
Comment (kcreddy) [inline packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml:349]: Add all `threat.indicator.file.hash.*` values into `related.hash`

Comment (mohitjha-elastic) [reply packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml:349]: All the hash values are already appended to the related.hash. `hash_sha1` was missing, added that in the recent commit.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼 lets wait for Dan's approval before merging

File: packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -347,25 +336,54 @@ processors:
...
+      field: threat.indicator.file.hash.sha1
+      tag: set_threat_indicator_file_hash_sha1
+      value: '{{{ti_crowdstrike.intel.value}}}'
+      if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.value != null && ctx.ti_crowdstrike.intel.type.contains('hash_sha1')
+  - set:
+      field: threat.indicator.file.hash.md5
+      tag: set_threat_indicator_file_hash_md5
       value: '{{{ti_crowdstrike.intel.value}}}'
       if: ctx.ti_crowdstrike?.intel?.type != null && ctx.ti_crowdstrike.intel.type.contains('hash_md5') && ctx.ti_crowdstrike?.intel?.value != null
+  - set:
```
Comment (efd6) [inline packages/ti_crowdstrike/data_stream/intel/elasticsearch/ingest_pipeline/default.yml:353]: Can we have test cases for these two changes as well? `threat.indicator.url.full` and `threat.indicator.ip`.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**